controllers/clusteroperator.go

```diff
@@ -133,40 +143,32 @@ func (r *ProvisioningReconciler) syncStatus(co *osconfigv1.ClusterOperator, cond
-		setStatusCondition(OperatorDisabled, osconfigv1.ConditionTrue, string(ReasonUnsupported), disabledMessage),
-	}
-
-	return r.syncStatus(co, conds)
-}
-
-// updateCOStatusDegraded updates the ClusterOperator's Degraded
-// degradedReason should contain the current reason for the Operator to be marked in that state
-func (r *ProvisioningReconciler) updateCOStatusDegraded(degradedReason string, detailedError string) error {
-	degradedMessage := "Operator is Degraded"
-	progressingMessage := "Operator is Degraded while Progressing"
-
-	co, err := r.getOrCreateClusterOperator()
-	if err != nil {
-		return err
-	}
-
-	conds := []osconfigv1.ClusterOperatorStatusCondition{
-		setStatusCondition(osconfigv1.OperatorDegraded, osconfigv1.ConditionTrue, degradedReason, degradedMessage),
-		setStatusCondition(osconfigv1.OperatorProgressing, osconfigv1.ConditionTrue, detailedError, progressingMessage),
-		setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionFalse, "", ""),
+	conds := defaultStatusConditions()
+	switch newReason {
+	case ReasonUnsupported:
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(OperatorDisabled, osconfigv1.ConditionTrue, string(newReason), msg))
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, string(ReasonEmpty), ""))
+	case ReasonSyncing:
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, string(newReason), msg))
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorProgressing, osconfigv1.ConditionTrue, string(newReason), progressMsg))
+	case ReasonComplete:
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, string(newReason), msg))
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorProgressing, osconfigv1.ConditionFalse, string(newReason), progressMsg))
+	case ReasonInvalidConfiguration, ReasonDeployTimedout:
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorDegraded, osconfigv1.ConditionTrue, string(newReason), msg))
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, string(ReasonEmpty), ""))
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorProgressing, osconfigv1.ConditionTrue, string(newReason), progressMsg))
+	case ReasonDeplymentCrashLooping:
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorDegraded, osconfigv1.ConditionTrue, string(newReason), msg))
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionFalse, string(newReason), msg))
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorProgressing, osconfigv1.ConditionFalse, string(newReason), progressMsg))
 	}
```

Comment:
Should the getOperandVersions() be implemented in some form (see https://github.com/openshift/cluster-baremetal-operator/pull/28/files#diff-e86bbec1cec0825c6c6f7f39bad704abR134) ?

Reply:
@sadasu I am happy to add it. Can I do it in a separate PR?

Reply:
https://github.com/openshift/cluster-baremetal-operator/pull/36